src/ViewHelpers/DebugViewHelper.php

```diff
@@ -144,27 +169,33 @@ protected static function dumpVariable($variable, $html, $level, $levels)
      */
     protected static function getValuesOfNonScalarVariable($variable)
     {
-        if ($variable instanceof \ArrayObject || is_array($variable)) {
-            return (array) $variable;
-        } elseif ($variable instanceof \Iterator) {
+        if ($variable instanceof \ArrayObject || \is_array($variable)) {
+            return (array)$variable;
```

Comment:
TYPO3 CMS uses a CGL where there is no space between cast and value. TYPO3 Fluid CGL by convention is to have a space.

The elseif to sequence of if statements here imho don't add to the readability and they make no difference to the execution. The rule set for number of elseif checks is not declared in this repository but again, if it's a reasonably low number there's no problem with it - but in larger quantities, a switch is preferred because it compiles to a hash map.

In this case, a better refactoring might have been a `switch (true) { case $variable instanceof Foobar: break; }` etc.

Reply:
> a switch is preferred because it compiles to a hash map.
> In this case, a better refactoring might have been a `switch (true) { case $variable instanceof Foobar: break; }` etc.

Actually, this only happens since PHP 7.2 and **only** if **all** cases are integers or strings. Otherwise it behaves like an `elseif` internally. Thus your example wouldn't trigger the optimization because it uses `instanceof` in the cases.

See: https://derickrethans.nl/php7.2-switch.html